server/api/game/controller.js

```diff
@@ -94,22 +161,48 @@ const moveGhosts = ({ ghosts, gridState, scatterStart }) => {
 
 const calculateNextGameState = () => {
   const {
-    ghosts, gridState, scatterStart,
+    ghosts, gridState, scatterStart, pacmanOne, pacmanTwo,
   } = gameState;
-  const { ghostsUpdated, moveGhostsCount } = moveGhosts(
-    { ghosts, gridState, scatterStart },
-  );
+  const { ghostsUpdated, moveGhostsCount } = moveGhosts({ ghosts, gridState, scatterStart });
+  const { pacmanUpdated: newPacmanOne } = movePacman({
+    pacman: pacmanOne, ghostsUpdated, gridState,
+  });
+  const { pacmanUpdated: newPacmanTwo } = movePacman({
+    pacman: pacmanTwo, ghostsUpdated, gridState,
+  });
+
+  dieIfOnGhost({ ghosts: ghostsUpdated, pacman: newPacmanOne });
+  dieIfOnGhost({ ghosts: ghostsUpdated, pacman: newPacmanTwo });
+
+  const {
+    score: pacmanOneScore,
+  } = eatFood({ pacman: pacmanOne, gridStateAfterPacmanMove: gridState });
+  const {
+    score: pacmanTwoScore,
+  } = eatFood({ pacman: pacmanTwo, gridStateAfterPacmanMove: gridState });
+  gameState.pacmanOne = { ...pacmanOne, ...newPacmanOne, score: pacmanOneScore };
+  gameState.pacmanTwo = { ...pacmanTwo, ...newPacmanTwo, score: pacmanTwoScore };
   gameState.moveGhostsCount = moveGhostsCount;
   gameState.ghosts = ghostsUpdated;
 };
 
+export const updateDirection = ({ playerId, direction }) => {
+  const { pacmanOne, pacmanTwo } = gameState;
```

Comment:
Store the playerId as the key. Repititive conditions are used.
or else, get the pacman object from the gamestate by matching with playerId.
or else, send from client side, whether, the client is pacman1 or pacman2